### PR TITLE
Add PIT2015 dataset

### PR DESCRIPTION
**Pull request by Victor0118**
@tuzhucheng Could you review this PR now? Thanks!

### CODE REVIEW

File: vdpwi/model.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def __init__(self, config):
         def make_conv(n_in, n_out):
             conv = nn.Conv2d(n_in, n_out, 3, padding=1)
             conv.bias.data.zero_()
-            nn.init.xavier_normal(conv.weight)
+            nn.init.xavier_normal_(conv.weight)
```
Comment (tuzhucheng) [inline vdpwi/model.py]: @daemon should be aware of this change. Does this mean we were not using `xavier_normal` initialization before since we are not setting the copy of the tensor to anything? If so this may affect the hyperparameter optimization and experimentation that was worked on.

File: vdpwi/__main__.py [hunk 2]
```diff
@@ -26,7 +26,7 @@ def evaluate_dataset(split_name, dataset_cls, model, embedding, loader, batch_si
     parser = argparse.ArgumentParser(description='PyTorch implementation of VDPWI')
     parser.add_argument('model_outfile', help='file to save final model')
     parser.add_argument('--dataset', help='dataset to use, one of [sick, msrvid, trecqa, wikiqa]', default='sick')
-    parser.add_argument('--word-vectors-dir', help='word vectors directory', default=os.path.join(os.pardir, os.pardir, 'Castor-data', 'embeddings', 'GloVe'))
+    parser.add_argument('--word-vectors-dir', help='word vectors directory', default=os.path.join(os.pardir, 'Castor-data', 'embeddings', 'GloVe'))
```
Comment (tuzhucheng) [inline vdpwi/__main__.py]: Shouldn't there be two `os.pardir` items, since the first parent directory if the root Castor repo and the next higher up parent directory is castorini?

File: datasets/pit2015.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+from torchtext.data.iterator import BucketIterator
+from torchtext.vocab import Vectors
+
+from datasets.castor_dataset import CastorPairDataset
+
+
+class PIT2015(CastorPairDataset):
+    NAME = 'pit2015'
+    NUM_CLASSES = 2
+    ID_FIELD = Field(sequential=False, dtype=torch.float32, use_vocab=False, batch_first=True)
```
Comment (tuzhucheng) [inline datasets/pit2015.py]: Maybe use torchtext v0.2.3 for now for the tensor type.

File: common/evaluators/pit2015_evaluator.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        rel_total = 0
+        pre_total = 0
+        for batch_idx, batch in enumerate(self.data_loader):
+            sent1, sent2 = self.get_sentence_embeddings(batch)
+            scores = self.model(sent1, sent2, batch.ext_feats, batch.dataset.word_to_doc_cnt, batch.sentence_1_raw, batch.sentence_2_raw)
+            prediction = torch.max(scores, 1)[1].view(batch.label.size()).data
+            gold_label = batch.label.data
+            n_dev_correct += (prediction == gold_label).sum().item()
+            acc_total += ((prediction == batch.label.data) * (prediction == 1)).sum().item()
+            total_loss += F.cross_entropy(scores, batch.label, size_average=False).item()
```
Comment (tuzhucheng) [inline common/evaluators/pit2015_evaluator.py]: I'm guessing you are using this with the VDPWI model, which uses `F.log_softmax`. If so you should be using NLL loss instead since cross_entropy does log softmax again.

From PyTorch documentation on NLLLoss:

```
Obtaining log-probabilities in a neural network is easily achieved by adding a LogSoftmax layer in the last layer of your network. You may use CrossEntropyLoss instead, if you prefer not to add an extra layer.
```

File: vdpwi/model.py (hunk 1 above)
Comment (Victor0118) [reply vdpwi/model.py]: https://pytorch.org/docs/stable/_modules/torch/nn/init.html

According to their doc, this is the correct usage?

Comment (Victor0118) [reply vdpwi/model.py]: In fact, it throws an warning and ask me to replace `xavier_normal` with `xavier_normal_`. That's why I make this change.

File: vdpwi/__main__.py (hunk 2 above)
Comment (Victor0118) [reply vdpwi/__main__.py]: https://github.com/castorini/Castor/blob/master/mp_cnn/__main__.py#L46

Then it is different from mpcnn. I think there should be not `castorini` directory.

File: datasets/pit2015.py (hunk 3 above)
Comment (Victor0118) [reply datasets/pit2015.py]: Yeah. You are right.

File: common/evaluators/pit2015_evaluator.py (hunk 4 above)
Comment (Victor0118) [reply common/evaluators/pit2015_evaluator.py]: Okay. To avoid this trouble, I want to delete loss from the evaluation metrics.

File: vdpwi/__main__.py (hunk 2 above)
Comment (tuzhucheng) [reply vdpwi/__main__.py]: Oh sorry, my bad, your chance of one `os.pardir` is correct since we run the script from the repo root.

File: vdpwi/model.py (hunk 1 above)
Comment (daemon) [reply vdpwi/model.py]: Fortunately, this does not affect hyperparameter optimization and experimentation. In < 0.4.0, `xavier_normal` was an in-place operation. In 0.4.0+, they deprecated `xavier_normal` and changed the name to `xavier_normal_`, which better reflects PyTorch convention of having underscore suffixes for in-place operations.

File: common/trainers/pit2015_trainer.py [hunk 5]
```diff
@@ -80,6 +80,6 @@ def train(self, epochs):
 
             prev_loss = avg_loss
             if scheduler is not None:
-                scheduler.step(accuracy)
+                scheduler.step(f1)
```
Comment (tuzhucheng) [inline common/trainers/pit2015_trainer.py]: Your scheduler is monitoring f1 but saving best model (dev score) is based on accuracy, is this as intended?
